server/src/game-implementations/games/ddr.ts

```diff
@@ -0,0 +1,332 @@
+import { GoalFmtPercent, GoalOutOfFmtPercent, GradeGoalFormatter } from "./_common";
+import db from "../../external/mongo/db";
+import { IsNullish } from "../../utils/misc";
+import { SessionAvgBest10For } from "../utils/session-calc";
+import {
+	DDR_GBOUNDARIES,
+	type Game,
+	GetGrade,
+	GetSpecificGPTConfig,
+	type integer,
+	type Playtype,
+} from "tachi-common";
+import type {
+	GPTGoalFormatters,
+	GPTGoalProgressFormatters,
+	GPTServerImplementation,
+	ScoreValidator,
+} from "game-implementations/types";
+import type { PBScoreDocument, SongDocument } from "tachi-common";
+
+interface PBScoreDocumentWithSong extends PBScoreDocument<"ddr:DP" | "ddr:SP"> {
+	song: SongDocument<"ddr">;
+	top?: number;
+}
+
+const FLARE_0_POINTS = [
+	145, 155, 170, 185, 205, 230, 255, 290, 335, 400, 465, 510, 545, 575, 600, 620, 635, 650, 665,
+];
+
+const DDR_GOAL_FMT: GPTGoalFormatters<"ddr:DP" | "ddr:SP"> = {
+	percent: GoalFmtPercent,
+
+	// don't want commas
+	score: (v) => `Get a score of ${v} on`,
+};
+
+const DDR_GOAL_OO_FMT: GPTGoalFormatters<"ddr:DP" | "ddr:SP"> = {
+	percent: GoalOutOfFmtPercent,
+	// don't insert commas or anything.
+	score: (m) => m.toString(),
+};
+
+const DDR_GOAL_PG_FMT: GPTGoalProgressFormatters<"ddr:DP" | "ddr:SP"> = {
+	percent: (pb) => `${pb.scoreData.percent.toFixed(2)}%`,
+
+	// 4519 -> "4519". Don't add commas or anything.
```

Comment:
Why not? I think for a game with numbers this large it should format the scores. Even the game itself displays `998273` as `998,273`.

See SDVX for how this is implemented.

Reply:
Probably because I'm french, I'm not used to commas in large numbers

Reply:
I think it delegates to an internationalised formatter in SDVX. If france doesn't format numbers like that then it won't for you.